async_service/asyncio.py

```diff
@@ -32,7 +32,15 @@ class FunctionTask(BaseFunctionTask):
     #
     async def run(self) -> None:
         try:
-            await self._async_fn(*self._async_fn_args)
+            # Name the task, to make it easier to identify which task is holding
+            #   onto the event loop for too long. Also, force an event loop cycle.
+            if sys.version_info >= (3, 8):
+                await asyncio.create_task(
```

Comment:
I think you want to do this somewhere else, because you are creating two asyncio tasks for every task.

https://github.com/ethereum/async-service/blob/25fd67a92c87e4d430ac49eeab47904e5c90a50a/async_service/asyncio.py#L309-L311

I think you want to do it here where we actually do the background scheduling.  You can do something like:

asyncio_task = asyncio.ensure_future(...)
if sys.version_info >= (3, 8):
    asyncio_task.set_name(task.name)
```